File: FrontEnd/SazonWareWA/SazonWareBO/GastoBO.cs
using SazonWareBO.ServicioWS;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SazonWareBO
{
    public class GastoBO : BaseBO
    {
        public BindingList<gasto> listarTodos()
        {
            gasto[] arreglo = WsGasto.listarGastos();
            if (arreglo == null)
            {
                return new BindingList<gasto>();
            }
            return new BindingList<gasto>(arreglo);
        }

        public int insertar(string titulo, string motivo, DateTime fecha, Double monto)
        {
            return this.WsGasto.insertarGasto( titulo, motivo, fecha, monto);
        }

        public int modificar(int idGasto, string titulo, string motivo, DateTime fecha, Double monto)
        {
            return this.WsGasto.modificarGasto(idGasto, titulo, motivo, fecha, monto);
        }

        public int eliminar(int idGasto)
        {
            return this.WsGasto.eliminarGasto(idGasto);
        }

        public gasto obtenerPorId(int idGasto)
        {
            return this.WsGasto.obtenerGastoPorId(idGasto);
        }

        public BindingList<gasto> buscarGastoFecha(int mes, int anhio)
        {
            gasto[] arreglo = this.WsGasto.listarGastosPorMes(mes,anhio);
            return new BindingList<gasto>(arreglo);
        }
    }
}

File: Backend/SoftGastoBO/src/pe/mibanco/soft/gasto/bo/GastoBO.java
package pe.mibanco.soft.gasto.bo;
import java.util.ArrayList;
import java.util.Date;
import pe.mibanco.soft.gasto.dao.GastoDAO;
import pe.mibanco.soft.gasto.daoImpl.GastoDAOImpl;
import pe.mibanco.soft.gasto.model.Gasto;


/**
 * Implementación del Business Object para la gestión de gastos
 */
public class GastoBO {
    private final GastoDAO gastoDAO;

    public GastoBO() {
        this.gastoDAO = new GastoDAOImpl();
    }

    public Integer insertar(String titulo, String motivo, Date fecha, Double monto) {
        Gasto gasto = new Gasto(null, titulo, motivo, fecha, monto);
        return gastoDAO.insertar(gasto);
    }

    public Integer modificar(Integer idGasto, String titulo, String motivo, Date fecha, Double monto) {
        Gasto gasto = new Gasto(idGasto, titulo, motivo, fecha, monto);
        return gastoDAO.modificar(gasto);
    }

    public ArrayList<Gasto> listarTodos() {
        return gastoDAO.listarTodos();
    }

    public Gasto obtenerPorId(Integer idGasto) {
        return gastoDAO.obtenerPorId(idGasto);
    }
    
    public Integer eliminar(Gasto gasto) {
        return gastoDAO.eliminar(gasto);
    }
    
    public ArrayList<Gasto> listarGastosPorMes(Integer mes, Integer anio) {
        GastoDAOImpl gastoDAO = new GastoDAOImpl();
        return gastoDAO.listarPorMes(mes, anio);
    }
}